ftlib/commlib/basic_commlib.py

```diff
@@ -1,7 +1,15 @@
 __version__ = "0.0.1"
+from typing import Set
 
 
-class BasicFramework:
+class BasicCommLib:
+    registered_apis: Set[str] = set()
+
+    @classmethod
+    def register_api(cls, api):
```

Comment:
What's the benefit of doing this instead of inheriting from abstract methods?

Reply:
When we trying to expose the low-level collective apis, without doing so, the user needs to call these apis via the communication library like `ftlib.commlib.broadcast`.

If we or other developers decorated the apis expected to be exposed with `BasicCommlib.register_api`, the `FTLib` class is able to recognize which apis are about to be exposed and will wrap these apis with the same [mechanism](https://github.com/caicloud/ftlib/blob/master/docs/imgs/ftlib_flow.png).

Reply:
Why not using `@classmethod` directly on the implementations of the sub-classes if they want to expose them as class methods? If they don't want to expose them, maybe they can prefix the function names by underscores.

Reply:
If `@classmethod` is defined in the subclass, other developers needs to understand how `registered_apis` should be modified with the decorator to fit the `FTLib`.

We defined the method in the base class so that other developers shall have no worry about how to implement the decorator.

Reply:
I see. Okay sounds good. Let's leave it as is for now and we can see if we can improve later on.